eCommerce platforms/Adobe Magento/src/code/local/Microsoft/Dfp/Helper/Data.php

```diff
@@ -0,0 +1,161 @@
+	}
+
+	public function log($msg)
+	{
+		if (Mage::getStoreConfigFlag('dfp/general/enablelog')) {
+			Mage::log($msg, null, 'MicrosoftDFP.log');
+		}
+	}
+
+	public function getFingerprintingSessionId()
+	{
+		$sessionId = Mage::getSingleton('core/session')->getFptDfpSessionId();
+		if (empty($sessionId)) {
+			$sessionId = $this->GUID();
+			Mage::getSingleton('core/session')->setFptDfpSessionId($sessionId);
+		}
+		return $sessionId;
+	}
+
+	public function sendUpdateAccount($payload, $correlationId = null)
+	{
+		return $this->invokeDFP('dfp/endpoints/updateaccount', $payload, $correlationId);
+	}
+
+	public function sendPurchase($payload, $correlationId = null)
+	{
+		return $this->invokeDFP('dfp/endpoints/purchase', $payload, $correlationId);
+	}
+
+	public function sendBankEvent($payload, $correlationId = null)
+	{
+		return $this->invokeDFP('dfp/endpoints/bankevent', $payload, $correlationId);
+	}
+
+	public function sendPurchaseStatus($payload, $correlationId = null)
+	{
+		return $this->invokeDFP('dfp/endpoints/purchasestatus', $payload, $correlationId);
+	}
+
+	public function sendRefund($payload, $correlationId = null)
+	{
+		return $this->invokeDFP('dfp/endpoints/refund', $payload, $correlationId);
+	}
+
+	public function sendChargeback($payload, $correlationId = null)
+	{
+		return $this->invokeDFP('dfp/endpoints/chargeback', $payload, $correlationId);
+	}
+
+	private function invokeDFP($configName, $payload, $correlationId)
+	{
+		$apiMethod = Mage::getStoreConfig($configName);
+		$correlationId = $correlationId ?: $this->GUID();
+		$jsonPayload = json_encode($payload);
+
+		$this->log("Payload " . $apiMethod . ": " . $jsonPayload);
+		$this->log("Correlation Id: " . $correlationId);
+
+		$accessToken = $this->getAccessToken();
+		if (empty($accessToken)) {
+			$this->log("Failed to get access token!");
+			throw new Exception("Failed to get access token!");
+		}
+
+		$this->log("Access Token: " . $accessToken);
+
+		$headers = array();
+		$headers[] = 'x-ms-correlation-id: ' . $correlationId;
+		$headers[] = 'Authorization: bearer ' . $accessToken;
+		$headers[] = 'Content-Type: application/json; charset=utf-8';
+
+		$baseurl = Mage::getStoreConfig('dfp/endpoints/baseurl');
+		$url = $this->buildUrl($baseurl, $apiMethod);
+
+		$curl = curl_init();
+		curl_setopt($curl, CURLOPT_POST, true);
+		curl_setopt($curl, CURLOPT_POSTFIELDS, $jsonPayload);
+		curl_setopt($curl, CURLOPT_URL, $url);
+		curl_setopt($curl, CURLOPT_RETURNTRANSFER, 1);
+		curl_setopt($curl, CURLOPT_HTTPHEADER, $headers);
+		$result = curl_exec($curl);
+		$httpCode = curl_getinfo($curl, CURLINFO_HTTP_CODE);
+		curl_close($curl);
+
+		$this->log("Response ($httpCode) : " . json_encode($result));
+		return $result;
+	}
+
+	private function getAccessToken()
+	{
+		$cacheId = 'dfp_accessToken';
+		if (false !== ($accessToken = Mage::app()->getCache()->load($cacheId))) {
```

Comment:
Any specific reason as to why this is backwards? Any reason to not split out the assignment and conditional into two different lines?

Reply:
Good call.  This was copy paste from a caching example.  Thanks for helping keep it clean.